library/Icingadb/Redis/VolatileStateResults.php

```diff
@@ -99,68 +106,92 @@ public function rewind(): void
     protected function applyRedisUpdates($rows)
     {
         $type = null;
-        $behaviors = null;
-
-        $keys = [];
-        $hostStateKeys = [];
-
         $showSourceGranted = $this->getAuth()->hasPermission('icingadb/object/show-source');
 
+        $getKeysAndBehaviors = function (State $state): array {
+            return [$state->getColumns(), $this->resolver->getBehaviors($state)];
+        };
+
         $states = [];
-        $hostStates = [];
+        $serviceHostStates = [];
```

Comment:
Why isn't everything in `$states`? I see no reason to differ between *hosts* and *service hosts*.